### PR TITLE
MONGOCRYPT-282 support chunked transfer encoding

### PR DESCRIPTION
**Pull request by kevinAlbs**
GCP HTTP responses used `Transfer-Encoding: chunked`. This adds support in the tiny kms-message HTTP parser.

### CODE REVIEW

File: kms-message/test/example-chunked-response.bin [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+X-Frame-Options: SAMEORIGIN
+X-Content-Type-Options: nosniff
+Alt-Svc: h3-Q050=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-27=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-T050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
+Accept-Ranges: none
+Vary: Origin,Accept-Encoding
+Connection: close
+Transfer-Encoding: chunked
+
+105
+{"access_token":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","expires_in":3599,"token_type":"Bearer"}
```
Comment (kevinAlbs) [inline kms-message/test/example-chunked-response.bin:18]: This is the form of a real GCP response, with the access token bytes replaced.

File: kms-message/test/test_kms_request.c [hunk 2]
```diff
@@ -964,6 +937,83 @@ kms_response_parser_test (void)
...
 }
 
+typedef struct {
+   const char *filepath;
+   const char *expected_body;
+   int max_to_read;
+   int expected_status;
+} parser_testcase_t;
+
+/* File should have \r\n line endings (use /etc/rewrite.py) */
```
Comment (markbenvenuto) [inline kms-message/test/test_kms_request.c:947]: Who is supposed to call rewrite.py? You can use git attributes to control how the file is checked out - https://www.git-scm.com/docs/gitattributes

File: kms-message/src/kms_response_parser.c [hunk 3]
```diff
@@ -254,6 +298,29 @@ kms_response_parser_feed (kms_response_parser_t *parser,
...
+            if (!parser->response->body) {
+               parser->response->body = kms_request_str_new ();
+            }
+            kms_request_str_append_chars (parser->response->body,
+                                          raw->str + parser->start,
+                                          parser->chunk_size);
+            curr = parser->start + parser->chunk_size + 2;
+            parser->start = curr;
+            if (parser->chunk_size == 0) {
+               printf ("done\n");
```
Comment (markbenvenuto) [inline kms-message/src/kms_response_parser.c:314]: Remove

File: kms-message/src/kms_response_parser.c [hunk 4]
```diff
@@ -101,6 +111,16 @@ _parse_int_from_view (const char *str, int start, int end, int *result)
    return ret;
 }
 
+static bool
+_parse_hex_from_view (const char *str, int start, int end, int *result)
```
Comment (markbenvenuto) [inline kms-message/src/kms_response_parser.c:115]: Just take length instead of (start, end)

File: kms-message/test/test_kms_request.c (hunk 2 above)
Comment (kevinAlbs) [reply kms-message/test/test_kms_request.c:947]: It's in etc, so just a development script for producing `\r\n` on non Windows. The default seems to preserve the `\r\n` (at least for .bin files).

File: kms-message/src/kms_response_parser.c (hunk 3 above)
Comment (kevinAlbs) [reply kms-message/src/kms_response_parser.c]: Gah, done.

File: kms-message/src/kms_response_parser.c (hunk 4 above)
Comment (kevinAlbs) [reply kms-message/src/kms_response_parser.c]: Done.

**Review by markbenvenuto — APPROVED** [review]

**Pull request merged**